### PR TITLE
Fix DDPG and SAC for eager execution

### PR DESCRIPTION
**Pull request by peabody124**
Trying to run DDPG/TD3/SAC would result in errors under Eager execution because
the OptimizerWrapper was not being used to compute the gradients. This PR updates
the code to run through this compatibility layer. Computing the gradients directly with
the optimizer in tf.compat.v1.AdamOptimizer should produce the same values. 

The prior broken and now fixed behavior can be confirmed with:
```
    rllib train --run SAC --env MountainCarContinuous-v0 --eager
    rllib train --run DDPG --env MountainCarContinuous-v0 --eager
    rllib train --run TD3 --env MountainCarContinuous-v0 --eager
```

DDPG also required adding an explicit cast to tf.float32 on the errors. Since there are
other casts hardcoded with this dtype, I don't believe this causes any loss of generality.
Finally the default SAC configuration is also updated to remove a deprecated field and
suppress the warning message.

### CODE REVIEW

**Comment by peabody124** [comment]
However, this breaks non-eager behavior, as the optimizer passed in does not have a  compute_gradients method. I can't quite see why this is, as I'm trying to follow the model from PPO (which works in both modes). Any guidance how to properly fix it for both models would be appreciated!

**Comment by ericl** [comment]
Can we wrap the two custom actor and critic optimizers for eager instead? Not sure the best way to do that though.

**Comment by sven1977** [comment]
Sorry, but we do have a tf2.x PR coming up in a week or so, which fixes DDPG and SAC for eager as well (also works for non eager). I think we should keep the different optimizers (critic 2x, actor, alpha) around. It could be useful to be able to configure them with different learning rates.

**Comment by peabody124** [comment]
This does retain the different learning rates since it still uses the separate optimizer when applying the gradients. The problem is that you don't really need the optimizer in the gradients function (at least for TF2) to compute the gradient in the ```gradients_fn``` and that is what compute_gradients is called on.

No worries if you guys have a better fix upcoming!

**Comment by sven1977** [comment]
Hey @peabody124 support for SAC and DDPG eager has been merged into master. We'll also merge full tf2.x support (rather than the current tf.compat.v1-mode) this weekend.
Thanks for your patience! Let us know, whether this works (or not) for you. :)

**Comment by peabody124** [comment]
Thanks for the updates. They DDPG work with eager on this now. Sorry it took me a bit to test it.

I don't think SAC in eager works, though. Using ```rllib train --env MountainCarContinuous-v0```
    SAC+Eager: reward stays at -35 after 16k steps
    SAC-Eager: reward at -35 @ ts=16k and -4 @ ts=100k
    DDPG+Eager: learns reasonably
    DDPG-Eager: learns reasonably

Also, I'm not sure if you guys are monitoring performance and how much this is in the rllib agents versus a pure TF issue but SAC with eager runs about 1/5 the rate of non-eager. It was slow enough at first I simply thought it wasn't working.

With DDPG eager is closer to 1/2 speed of non-eager (and seems more sample efficient with MountainCarContinuous so that is less apparent).

This is all CPU only.

I'll handle this PR, though, since you've fixed the general eager issue and actually in testing again my fix also did not get SAC to learn -- just to run (like the current HEAD)
